src/LumiSpecHomoCAL_geo.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ Written by - Aranya Giri
+ University of Houston
+ Start Date - 10/31/2022
+ Homogeneous PbWO4 (EM Calorimeter) Pair Spectrometer
+ Note : Solid, Logical & Physical Volumes are the three component of detector,
+ The following abbriviations will reflect the proper usage of these 3.
+*/
+
+// Essential Header Files
+#include "DD4hep/DetFactoryHelper.h"
+//#include "GeometryHelpers.h"
+#include <XML/Helper.h>
+#include <algorithm>
+#include <iostream>
+#include <math.h>
+#include <tuple>
+
+//Additional Header Files
+
+using namespace std;
+using namespace dd4hep;
+//using Point = ROOT::Math::XYPoint;
+
+//Definition of function to build the modules
+tuple<Volume, Position> build_specHomoCAL_module(const Detector& description, const xml::Component& mod_x, SensitiveDetector& sens);
+
+//Driver Function
+static Ref_t create_detector(Detector& description, xml_h e, SensitiveDetector sens)
+{
+  sens.setType("calorimeter");
+
+  xml_det_t 	x_det 		= 	e;
+  //xml_dim_t 	x_dim 		= 	x_det.dimensions();
+  xml_dim_t	x_pos		= 	x_det.position();
+  xml_dim_t	x_rot		= 	x_det.rotation();
+  xml_comp_t    x_mod           =       x_det.child( _Unicode(module) );
+  string        det_name	= 	x_det.nameStr();
+  int		det_ID		=	x_det.id();
+
+  // Create main detector element to be returned at the end
+  DetElement    det( det_name, det_ID );
+
+  // Mother volume
+  Volume        motherVol = description.pickMotherVolume( det );
+
+  // Detector assembly
+  Assembly      assembly( det_name );
+  assembly.setVisAttributes( description.invisible() );
+
+  //Create Modules
+  int nxy = 8;
+  int sector_id = x_mod.attr<int>( _Unicode( sector ) );
+  int mod_id = 0;
+
+  auto [modVol, modSize] = build_specHomoCAL_module(description, x_mod, sens);
+  double xypos0 = -(nxy*modSize.x())/2.0 + modSize.x()/2.0 ;
+
+  for(int ix=0; ix< nxy; ix++){
+    for(int iy=0; iy< nxy; iy++){
+
+      double 	mod_pos_x 	= xypos0 + ix*modSize.x();
+      double	mod_pos_y 	= xypos0 + iy*modSize.y();
+      double 	mod_pos_z 	= 0.0*cm;
+
+      PlacedVolume modPV = assembly.placeVolume(
+          modVol, Position( mod_pos_x, mod_pos_y, mod_pos_z ) );
+
+      modPV.addPhysVolID( "sector", sector_id ).addPhysVolID( "module", mod_id );
```

Comment:
What is your thinking regarding modules and sectors? I think you'll still need the row and column id's otherwise all of these PlacedVolumes will have the same volumeID.

Reply:
Sector will distinguish "top" vs "bottom".  Module will distinguish each of the 64=8x8 modules.  
Then there's the system ID.  I think each PbWO4 crystall has a unique physical volume ID now, right?

Reply:
I'd somehow missed the mod_id++ below.

Reply:
I would suggest having the same readout label to distinguish top/bottom for the tracker and calorimeter. The system ID could be the same for both calorimeters if the sector was put in loop, as it is in the tracker.

Reply:
Agreed.  I changed the readout scheme accordingly.